### PR TITLE
Add check for KeyUsage bit 

### PR DESCRIPTION
**Pull request by jyao1**
Fix https://github.com/DMTF/libspdm/issues/2495

### CODE REVIEW

**Comment by steven-bellock** [comment]
Requester needs to check `KEY_EXCHANGE_RSP.SlotIDParam`, if it exists.

**Comment by jyao1** [comment]
That is good point. I think we need SPDM spec may need to clarify that.
The `SlotID` usage for `FINISH` should be covered by `KeyExUse`.

File: library/spdm_requester_lib/libspdm_req_key_exchange.c [hunk 1]
```diff
@@ -536,6 +536,19 @@ static libspdm_return_t libspdm_try_send_receive_key_exchange(
                 status = LIBSPDM_STATUS_INVALID_MSG_FIELD;
                 goto receive_done;
             }
+
+            /* relax the key_usage_bit check for slot 0, because the responder might no know the key usage yet */
```
Comment (steven-bellock) [inline library/spdm_requester_lib/libspdm_req_key_exchange.c:540]: This is a bug in the specification.

Comment (steven-bellock) [inline library/spdm_requester_lib/libspdm_req_key_exchange.c:540]: If the Responder has chosen `SPDM_KEY_EXCHANGE_RESPONSE_MUT_AUTH_REQUESTED` then it should already know the Requester's multi-key restrictions, else the Responder would issue `GET_DIGESTS`. So the `(*req_slot_id_param != 0x0)` is not needed.

Comment (jyao1) [reply library/spdm_requester_lib/libspdm_req_key_exchange.c:540]: This field is mandatory.

Without the check, how do you know if this field should be used or not?

Comment (steven-bellock) [reply library/spdm_requester_lib/libspdm_req_key_exchange.c:540]: `SPDM_KEY_EXCHANGE_RESPONSE_MUT_AUTH_REQUESTED` means the Responder wants to skip the encapsulated flow and go straight to `FINISH`. If Responder specifies slot 0 then it means slot 0 and the Requester should check if its slot 0 key can do key exchange.

Comment (jyao1) [reply library/spdm_requester_lib/libspdm_req_key_exchange.c:540]: Ah, I see. Agree.

**Review by steven-bellock — APPROVED** [review]

**Pull request merged**